### PR TITLE
feat: add validation workflow from MC & add 2023 files

### PR DESCRIPTION
**Pull request by Ming-Yan**
**New content**
- testfile/notebook : add new notebook to extract ROC from validation workflow
- workflow/utils: add validation workflows & btag mask
-  doc/condor:add new modules `condor/missingFiles.py`
- data, utils: include 2023 conditions

**Bug fix**
-  ci: change certificate file permission
- script: fix template file name
- script: fix plot code
- workflow : correct syst setup

**Other changes** 
- util : change the implementation of jet veto

### CODE REVIEW

**Comment by Ming-Yan** [comment]
@adeiorio can you please take a look whether the changes of the condor script included?

**Comment by adeiorio** [comment]
> @adeiorio can you please take a look whether the changes of the condor script included?

Hi @Ming-Yan the changes in the condor script are ok. Thanks!

**Review by uttiyasarkar — COMMENTED** [review]
The following samples still seem to be missing in the JSON metadata:

SingleElectron:
data_Summer22_2022_e_BTV_Run3_2022_Comm_MINIAODv4_NanoV12.json
data_Summer22EE_2022_e_BTV_Run3_2022_Comm_MINIAODv4_NanoV12.json
data_Summer23_2023_e_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json
data_Summer23BPix_2023_e_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json

Naming of one file should be updated:
data_Summer23_mu_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json ->
data_Summer23_2023_mu_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json

File: .github/workflows/BTA_workflow.yml
[the remark this answers is not in the document]
Comment (Ming-Yan) [reply .github/workflows/BTA_workflow.yml:1]: this is independent for which environment of the user, this is for the ci

**Comment by Ming-Yan** [comment]
> The following samples still seem to be missing in the JSON metadata:
> 
> SingleElectron: data_Summer22_2022_e_BTV_Run3_2022_Comm_MINIAODv4_NanoV12.json data_Summer22EE_2022_e_BTV_Run3_2022_Comm_MINIAODv4_NanoV12.json data_Summer23_2023_e_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json data_Summer23BPix_2023_e_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json


There were no dedicated  electron workflows 

> Naming of one file should be updated: data_Summer23_mu_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json -> data_Summer23_2023_mu_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json

Done

Regarding to the mamba environment for the workflows, this is for the git actions :) should independent from user enviroment :)

**Comment by Ming-Yan** [comment]
change the recommendation to 2023.
clarification of json given 
emu-> stands for mueg dataset, so DY should runs on data_Summer23_2023_mu_BTV_Run3_2023_Comm_MINIAODv4_NanoV12.json
and for ctag_ttdilep_sf this is for cSF with dileptonic selection, we can use either electron/muon/mueg dataset. so this is correct as well

**Comment by uttiyasarkar** [comment]
Thanks, the PR looks good now. Merging it to the master branch.

**Pull request merged**